### ISSUE TITLE
Add WeCom syncer to synchronize users and groups between Casdoor and WeCom

### ISSUE THREAD

**Issue by hsluoyz** [casdoor/casdoor#2360]
What's WeCom: https://work.weixin.qq.com/

What's Casdoor's syncer: https://casdoor.org/docs/syncer/overview/

The available Go SDKs to use, like:

- https://github.com/go-laoji/wecom-go-sdk
- https://github.com/xen0n/go-workwx

What needs to be synced?

- Users:
  - Casdoor users: https://door.casdoor.com/organizations/built-in/users
  - WeCom users: https://github.com/go-laoji/wecom-go-sdk#:~:text=%E9%80%9A%E8%AE%AF%E5%BD%95%E7%AE%A1%E7%90%86-,%E6%88%90%E5%91%98%E7%AE%A1%E7%90%86,-%E5%88%9B%E5%BB%BA%E6%88%90%E5%91%98

- Groups:
  - Casdoor groups: https://door.casdoor.com/trees/built-in
  - WeCom groups: https://github.com/go-laoji/wecom-go-sdk#:~:text=%E6%88%90%E5%91%98ID%E5%88%97%E8%A1%A8-,%E9%83%A8%E9%97%A8%E7%AE%A1%E7%90%86,-%E5%88%9B%E5%BB%BA%E9%83%A8%E9%97%A8
 
![image](https://github.com/casdoor/casdoor/assets/3787410/4624ebf0-ff66-439c-bd06-2d5ec5874d5d)

**Comment by hsluoyz** [comment]
@UsherFall

**Comment by jason-lee-x** [comment]
Following up on the **Groups** part of this issue after verifying the current implementation and its history.

This part of the original requirement is still not implemented on current `master` (`7325999791deaeab538f86082d70858d0ab4d773`) or v3.28.0:

- `WecomUser.Department` is decoded, but `wecomUserToOriginalUser()` always sets `Groups: []string{}`.
- `WecomSyncerProvider.GetOriginalGroups()` is explicitly a TODO and returns an empty slice.
- `WecomSyncerProvider.GetOriginalUserGroups()` is also a TODO and returns an empty slice.
- Manual `RunSyncer()` currently calls only `syncUsers()`; scheduled sync does call `syncGroups()`, but the WeCom provider returns no groups.

I also checked the history to avoid misclassifying this:

- The merged PR #2377 did include WeCom department-to-Casdoor-group conversion and user group mapping.
- Issue #4328 later states that the old WeCom syncer had been reverted; its replacement PR #4329 was intentionally scoped to syncing users into Casdoor.
- PR #4962 subsequently added the generic group-sync interface for Google Workspace and left WeCom group methods as TODOs.

Reproduction on a WeCom tenant with a non-empty multi-level department tree: user synchronization completes successfully, but Casdoor creates zero groups and imported users have empty group/affiliation values.

Expected behavior: map WeCom departments (`id`, `name`, `parentid`) to the Casdoor Group tree (`Name`/`DisplayName`/`ParentId`), and map each user's `department[]` (or at least `main_department`) to `User.Groups`.

Could the Groups portion of this issue be reopened, or should it be tracked in a dedicated follow-up issue?
